presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -165,38 +167,62 @@ public static PlanNodeStatsEstimate addStatsAndCollapseDistinctValues(PlanNodeSt
         return addStats(left, right, StatisticRange::addAndCollapseDistinctValues);
     }
 
-    private static PlanNodeStatsEstimate addStats(PlanNodeStatsEstimate left, PlanNodeStatsEstimate right, RangeAdditionStrategy strategy)
+    public static PlanNodeStatsEstimate addStatsAndIntersect(PlanNodeStatsEstimate left, PlanNodeStatsEstimate right)
+    {
+        if (left.isOutputRowCountUnknown() || right.isOutputRowCountUnknown()) {
+            return PlanNodeStatsEstimate.unknown();
+        }
+
+        PlanNodeStatsEstimate.Builder statsBuilder = PlanNodeStatsEstimate.builder();
+        double estimatedRowCount = Math.min(left.getOutputRowCount(), right.getOutputRowCount());
+
+        List<Double> allRowCounts = buildVariableStatistics(
+                left, right, statsBuilder, estimatedRowCount, StatisticRange::intersect);
+
+        double preciseRowCount = allRowCounts.stream().reduce(estimatedRowCount, Double::min);
+        return statsBuilder.setOutputRowCount(preciseRowCount).build();
+    }
+
+    private static PlanNodeStatsEstimate addStats(PlanNodeStatsEstimate left, PlanNodeStatsEstimate right,
```

Comment:
style note: we either have all the arguments on one line, or if that's too long, we split it to one per-line.

Reply:
Okay. I did not know that. I see several other pre-existing functions in this file were not following this convention. I will update the new function as well as other existing ones.